iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java

```diff
@@ -142,19 +142,23 @@ public File file(Optional<List<Statement>> statements, Optional<SyntaxToken> spa
     return new FileImpl(statements.or(emptyList()), eof);
   }
 
-  // TODO SONARIAC-951 Put in place decorator
-  public TypeDeclaration typeDeclaration(SyntaxToken keyword, Identifier name, SyntaxToken equ, StringLiteral typeExpression) {
-    return new TypeDeclarationImpl(keyword, name, equ, typeExpression);
+  public TypeDeclaration typeDeclaration(Optional<List<Decorator>> decorators, SyntaxToken keyword, Identifier name, SyntaxToken equ, StringLiteral typeExpression) {
+    return new TypeDeclarationImpl(decorators.or(emptyList()), keyword, name, equ, typeExpression);
   }
 
-  // TODO SONARIAC-957 Put in place decorator
-  public OutputDeclaration outputDeclaration(SyntaxToken keyword, Identifier name, Identifier type, SyntaxToken equ, Expression expression) {
-    return new OutputDeclarationImpl(keyword, name, type, equ, expression);
+  public OutputDeclaration outputDeclaration(Optional<List<Decorator>> decorators, SyntaxToken keyword, Identifier name, Identifier type, SyntaxToken equ, Expression expression) {
+    return new OutputDeclarationImpl(decorators.or(emptyList()), keyword, name, type, equ, expression);
   }
 
-  // TODO SONARIAC-957 Put in place decorator
-  public OutputDeclaration outputDeclaration(SyntaxToken keyword, Identifier name, SyntaxToken resource, InterpolatedString type, SyntaxToken equ, Expression expression) {
-    return new OutputDeclarationImpl(keyword, name, resource, type, equ, expression);
+  public OutputDeclaration outputDeclaration(
+    Optional<List<Decorator>> decorators,
+    SyntaxToken keyword,
+    Identifier name,
+    SyntaxToken resource,
+    InterpolatedString type,
+    SyntaxToken equ,
+    Expression expression) {
+    return new OutputDeclarationImpl(decorators.or(emptyList()), keyword, name, resource, type, equ, expression);
```

Comment:
I would prefer for all the decorators to not store an emptyList, but instead use `decorators.orNull()`.
Otherwise we would store a lot of emptyLists.